### PR TITLE
Enable pylint for ansible-freeipa roles.

### PR DESCRIPTION
**Pull request by rjeffman**
Pylint was not being applied to ansible-freeipa roles
in upstream CI or in utility scripts. This patch enables
pylint for roles in both scenarios.

### CODE REVIEW

File: roles/ipabackup/library/ipabackup_get_backup_dir.py [hunk 1]
```diff
@@ -61,7 +61,7 @@
 
 def main():
     module = AnsibleModule(
-        argument_spec=dict(),
+        argument_spec={},
```
Comment (t-woerner) [inline roles/ipabackup/library/ipabackup_get_backup_dir.py:64]: Is this change needed for pylint?

Comment (rjeffman) [reply roles/ipabackup/library/ipabackup_get_backup_dir.py:64]: Yes, it is needed. An alternative is to disable this check.

Comment (rjeffman) [reply roles/ipabackup/library/ipabackup_get_backup_dir.py:64]: So, our pre-commit hook uses pylint 2.10.2, but Fedora 35 provides 2.9.6 (latest upstream is 2.12.2).

If we stick to Fedora's version, this change is not needed. It will only become necessary when 2.10 series land on Fedora.

File: roles/ipaserver/library/ipaserver_test.py [hunk 2]
```diff
@@ -942,7 +943,7 @@ def main():
     else:
         realm_name = options.realm_name.upper()
 
-    argspec = inspect.getargspec(validate_domain_name)
+    argspec = inspect.getfullargspec(validate_domain_name)
```
Comment (t-woerner) [inline roles/ipaserver/library/ipaserver_test.py:946]: Why is there a change from getargspec to getfullargspec?

Comment (rjeffman) [reply roles/ipaserver/library/ipaserver_test.py]: It shouldn't. Fixed.

**Review by t-woerner — APPROVED** [review]
LGTM

**Review by t-woerner — APPROVED** [review]
LGTM
